### PR TITLE
Get rid of @ts-ignore (iteration 1 or many more to go)

### PR DESCRIPTION
**Pull request by dnafication**
- Removed the silentNpm function as it was not being used anywhere
- Type guard for childProcess.stdout in process.ts
- fix type errors by adding type guards
- fix typos
- Attempted to remove some ts-ignore comments

### CODE REVIEW

**Comment by dnafication** [comment]
Strange! the build passes locally without any issues.
```
packages/server/express build$ pnpm run build:js && pnpm run build:types
│ > @verdaccio/server@7.0.0-next.3 build:js /Users/dina/code/personal/verdaccio/packages/server/express
│ > babel src/ --out-dir build/ --copy-files --extensions ".ts,.tsx" --source-maps
│ Successfully compiled 4 files with Babel (266ms).
│ > @verdaccio/server@7.0.0-next.3 build:types /Users/dina/code/personal/verdaccio/packages/server/express
│ > tsc --emitDeclarationOnly -p tsconfig.build.json
└─ Done in 3.5s
```

Going to rerun the failed job.

**Comment by dnafication** [comment]
Seems to run fine locally on my mac but the build is failing in Github Actions with the error below:

```
Error: packages/server/express build: src/server.ts(19,27): error TS2307: Cannot find module '@verdaccio/web' or its corresponding type declarations.
packages/server/express build:  ELIFECYCLE  Command failed with exit code 2.
packages/server/express build: Failed
/home/runner/work/verdaccio/verdaccio/packages/server/express:
 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @verdaccio/server@7.0.0-next.3 build: `pnpm run build:js && pnpm run build:types`
Exit status 1
 ELIFECYCLE  Command failed with exit code 1.
Error: Process completed with exit code 1.
```

I'll try to recreate the GHA env locally to see if I can get more details

**Comment by juanpicado** [comment]
> Seems to run fine locally on my mac but the build is failing in Github Actions with the error below:
> 
> ```
> Error: packages/server/express build: src/server.ts(19,27): error TS2307: Cannot find module '@verdaccio/web' or its corresponding type declarations.
> packages/server/express build:  ELIFECYCLE  Command failed with exit code 2.
> packages/server/express build: Failed
> /home/runner/work/verdaccio/verdaccio/packages/server/express:
>  ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @verdaccio/server@7.0.0-next.3 build: `pnpm run build:js && pnpm run build:types`
> Exit status 1
>  ELIFECYCLE  Command failed with exit code 1.
> Error: Process completed with exit code 1.
> ```
> 
> I'll try to recreate the GHA env locally to see if I can get more details

yeah fails on master, I don't know why I haven't change anything meaningful, I'll check

**Comment by juanpicado** [comment]
please rebase and use `corepack prepare --activate pnpm@8.9.0` , seems `8.9.1` is breaking something.

**Comment by dnafication** [comment]
Thanks @juanpicado! I had never used `corepack` before. I learned about it here. One big advantage of contributing to open source is learning and gaining new knowledge.

File: packages/store/src/storage.ts [hunk 1]
```diff
@@ -983,7 +983,7 @@ class Storage {
     const localPackage = await this.getPackageManifest({
       name,
       requestOptions,
-      uplinksLook: false,
```
Comment (juanpicado) [inline packages/store/src/storage.ts:986]: here `false` is intended, check the variable name, `localPackage`

Comment (dnafication) [reply packages/store/src/storage.ts:986]: I'll revert it.

**Comment by dnafication** [comment]
@juanpicado sorry to bother but this has been resolved ☝🏼

File: packages/api/src/publish.ts [hunk 2]
```diff
@@ -196,10 +196,10 @@ export function publishPackage(storage: Storage): any {
         requestOptions: {
           host: req.hostname,
           protocol: req.protocol,
-          // @ts-ignore
-          headers: req.headers,
+          headers: req.headers as { [key: string]: string },
           username,
         },
+        uplinksLook: false,
```
Comment (juanpicado) [inline packages/api/src/publish.ts:202]: LGTM so far, just checking what's the side effects of this one, seems not even used inside, maybe is not even mandatory.

**Pull request merged**